xbmc/addons/Addon.cpp

```diff
@@ -264,6 +264,54 @@ bool CAddon::SettingsLoaded(AddonInstanceId id /* = ADDON_SETTINGS_ID */) const
   return addonSettings && addonSettings->IsLoaded();
 }
 
+namespace
+{
+
+struct CacheItem
+{
+  time_t modified;
+  CXBMCTinyXML xml;
+};
+
+using FilenameXMLCache = std::unordered_map<std::string, CacheItem>;
+
+FilenameXMLCache XMLFileCache{};
+
+CXBMCTinyXML* loadXMLFile(const std::string& id, const std::string& xmlFilename)
```

Comment:
Can this return `const CXBMCTinyXML*`? The loaded XML shouldn't need to be modified.

Reply:
You are correct. Done.